pkg/pillar/types/downloadertypes.go

```diff
@@ -4,15 +4,29 @@
 package types
 
 import (
-	log "github.com/sirupsen/logrus"
 	"time"
+
+	log "github.com/sirupsen/logrus"
 )
 
+type RktCredentials struct {
```

Comment:
This is interesting, why does downloader know its container credentials, why cannot we use common user/name password location, does there different sturcture for SFTP ?

Reply:
For other types, we are directly passing it to the command. For rkt, we need to put that into a json file and pass the file. Hence doing it this way.